File: test/ModularPipelines.UnitTests/Attributes/WindowsOnlyTestAttribute.cs
namespace ModularPipelines.UnitTests.Attributes;

public class WindowsOnlyTestAttribute() : SkipAttribute("Windows only test")
{
    public override Task<bool> ShouldSkip(TestRegisteredContext context)
    {
        return Task.FromResult(!OperatingSystem.IsWindows());
    }
}

File: test/ModularPipelines.UnitTests.FSharp/Attributes/WindowsOnlyTestAttribute.fs
namespace ModularPipelines.UnitTests.FSharp.Attributes

open System
open System.Threading.Tasks
open TUnit.Core

type WindowsOnlyTestAttribute() =
    inherit SkipAttribute("Windows only test")
    override _.ShouldSkip(_: TestRegisteredContext) =
        Task.FromResult(not (OperatingSystem.IsWindows()))
